### DISCUSSION TITLE
Ecosystem Critical Projects?

### DISCUSSION THREAD

**Discussion by Relequestual** [json-schema-org/TSC#5] [TSC Discussions]
The status of this TSC decision is tracked using https://github.com/json-schema-org/TSC/issues/6

This Issue also details the stage of the discussion and what actions are required to progress to the next stage.

### Background

During our [in person JSON Schema summit](https://github.com/orgs/json-schema-org/discussions/391) last year (2023), one of the items on the agenda was how we allow existing projects into the JSON Schema Github organization.
We concluded that rather than ingesting existing projects, we would look to support and enable them where they were.

> Decision: What better fits with our ecosystem is to define a program to engage with implementers, offering resources and support and allowing them to keep the property. For JSON Schema to be successful we need implementers to be successful.

We created an Issue to track strategy for engaging implementers: https://github.com/json-schema-org/community/issues/412. And, while this is good, it didn't reflect the need to support specific critical projects.

Since then, we have had some discussions about a "critical projects" program, which is slightly different. Discussion resurfaced on [a community call on 2024-01-08](https://github.com/json-schema-org/community/issues/566). And, @benjagm opened [an Issue](https://github.com/json-schema-org/community/issues/521) to kickstart discussion and drive this work forward.

### What do we need to decide on?

- Is the phrasing "ecosystem critical projects" a good representation of the target projects?
- How do we identify / classify "critical projects"?
- What do critical projects need most?
- How can we best enable and support critical projects?
- Will this be exclusivly for existing projects?
- Should we set general conditions and periodic reviews?
- Should we also be able to set project specific conditions or objectives?

This is not an exhaustive list and more questions may be added.
(Please also feel free to suggest better phrasing of questions that might lead to more open discussion.)

---
While this Discussion is labelled `tsc-decision` and is associated with a consensus stage per our governance process, both the community and TSC are welcome to provide their thoughts and comments on these matters.

**Comment by gregsdennis** [comment]
Is the phrasing "ecosystem critical projects" a good representation of the target projects?

**Comment by gregsdennis** [comment]
How do we identify / classify "critical projects"?

**Comment by gregsdennis** [comment]
What do critical projects need most?

**Reply by jviotti** [reply]
I think this is the key question worth answering that would drive the rest. But it is a tricky question, and I don't even know what the answer would be for my own projects. Are projects getting stuck because of knowledge? Because of time?

**Reply by Relequestual** [reply]
> I don't even know what the answer would be for my own projects. Are projects getting stuck because of knowledge? Because of time?

If we can get a few implementers together, I'd be happy to arrange a workshop session to try and collectivly work through this sort of question. Thumbs up this comment to signal your interst in this!

**Reply by thclark** [reply]
Very happy to join a workshop on this. A JSONSchema repository app, 'Strands', will come into public beta in the next couple of months (@relequestual and @jviotti I finally got a bit of funding to kick it over the finish line - or I should probably say "start line"!!).

I'll need:
- very early feedback from a small core community on basic functionality
- ability to poll the community for what key features are most important to add next
- some publicity or engagement with increasingly wide communities over the months subsequent to release as we soft-launch

As a more general answer, a couple of years ago what would have been useful (and I suppose still would be) was a much clearer idea of market size - ie number of users and organisations using JSONSchema (I had to wild guess), and traffic sizes I might expect.

In general, another thing implementers need help with is technical support on the subtleties of JSONSchema which the slack community is very good at answering, and better documentation and other such things that you already know about and are on the roadmap / under development in GSoC.

**Comment by gregsdennis** [comment]
How can we best enable and support critical projects?

**Reply by omissis** [reply]
Hi! go-jsonschema maintainer here :) As a person who's not super knowledgeable about json-schema and maintaining the project in his spare time, maybe something like a maintainer guide (in the shape of a site, or a bunch of markdown files) could be helpful.
A few things that might be mentioned there:
- The existence of a language-agnostic test suite: https://github.com/json-schema-org/JSON-Schema-Test-Suite
- Known issues or problematic bits of the implementation authors encountered in their implementation (eg: file references, or lack thereof, security, ambiguities, etc)
- A collection of resources to learn from others (eg: meetups, recordings, articles, etc)

I hope it helps! And thank you for the amazing job you folks are doing, jsonschema is possibly the most welcoming community I have ever participated to! 🙏

**Reply by jviotti** [reply]
I really like the second and third points and would love to participate on those. For example, I wrote the lexical/dynamic scope blog post as a way to do the second.

**Reply by jviotti** [reply]
Also on number 2 and 3, we are doing cool research on GSoC for the upgrade/downgrade project and we'll have a series of research documents as a result (see https://github.com/json-schema-org/upgrade-downgrade-rules/blob/main/research/format.markdown that studies `format` over time). Hopefully we can convert some of those to blog posts or something like that.

**Reply by Relequestual** [reply]
@omissis thanks for sharing your feedback on this.
This would be a more general way to support and enable maintainers for sure.
We do have an [implementers section of the site](https://json-schema.org/implementers), but it doesn't have all the things you suggest.
I'll make sure we capture your feedback!

I'm thinking about more direct / hands-on ways in which we can support such projects.
I'm also thinking about more established projects as opposed to new projects. I can see you are a more established project.

Does this spark any further ideas?

**Reply by thclark** [reply]
I think my comments on this question are the same as on the previous question ("What do critical projects need most?") which is similar in nature.

**Comment by gregsdennis** [comment]
Will this be exclusivly for existing projects?

**Reply by gregsdennis** [reply]
I wouldn't expect so.  I imagine any new project that comes along and satisfies whatever conditions we put in place could be considered a "critical" project.

I also expect existing projects to eventually lose momentum, as is the way of open source.

This will likely be a revolving door of projects, hopefully over the long term only, though.

**Comment by gregsdennis** [comment]
Should we set general conditions and periodic reviews?

**Comment by gregsdennis** [comment]
Should we also be able to set project specific conditions or objectives?
